tree-sitter.d.ts

```diff
@@ -140,11 +167,52 @@ declare module "tree-sitter" {
       readonly setProperties: any[];
       readonly assertedProperties: any[];
       readonly refutedProperties: any[];
+      readonly matchLimit: number;
+
+      constructor(
+        language: any,
+        source: string | Buffer,
+      );
+
+      matches(
+        rootNode: SyntaxNode,
+        options?: {
+          startPosition?: Point;
+          endPosition?: Point;
+          startIndex?: number;
+          endIndex?: number;
+          matchLimit?: number;
+          maxStartDepth?: number;
+        }
+      ): QueryMatch[];
+      captures(
+        rootNode: SyntaxNode,
+        options?: {
+          startPosition?: Point;
+          endPosition?: Point;
+          startIndex?: number;
+          endIndex?: number;
+          matchLimit?: number;
+          maxStartDepth?: number;
+        }
+      ): QueryCapture[];
+      disableCapture(captureName: string): void;
+      disablePattern(patternIndex: number): void;
+      isPatternGuaranteedAtStep(byteOffset: number): boolean;
+      isPatternRooted(patternIndex: number): boolean;
+      isPatternNonLocal(patternIndex: number): boolean;
+      startIndexForPattern(patternIndex: number): number;
+      didExceedMatchLimit(): boolean;
+    }
 
-      constructor(language: any, source: string | Buffer);
+    export class LookaheadIterator implements IterableIterator<number> {
+      readonly currentSymbol: number;
+      readonly currentSymbolName: string;
 
-      matches(rootNode: SyntaxNode, startPosition?: Point, endPosition?: Point): QueryMatch[];
-      captures(rootNode: SyntaxNode, startPosition?: Point, endPosition?: Point): QueryCapture[];
+      reset(language: any, state: number): boolean;
+      resetState(state: number): boolean;
+      next(): number;
```

Comment:
I get this type error trying to use this code:

```
../third-party/node-tree-sitter/tree-sitter.d.ts:214:7 - error TS2416: Property 'next' in type 'LookaheadIterator' is not assignable to the same property in base type 'IterableIterator<number>'.
  Type '() => number' is not assignable to type '(...args: [] | [undefined]) => IteratorResult<number, any>'.
    Type 'number' is not assignable to type 'IteratorResult<number, any>'.

214       next(): number;
          ~~~~
```

If I change this to

```suggestion
      next(): IteratorResult<number>;
```

then it complains that

```
../third-party/node-tree-sitter/tree-sitter.d.ts:208:18 - error TS2420: Class 'LookaheadIterator' incorrectly implements interface 'IterableIterator<number>'.
  Property '[Symbol.iterator]' is missing in type 'LookaheadIterator' but required in type 'IterableIterator<number>'.

208     export class LookaheadIterator implements IterableIterator<number> {
                     ~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.es2015.iterable.d.ts:53:5
    53     [Symbol.iterator](): IterableIterator<T>;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    '[Symbol.iterator]' is declared here.


Found 1 error in ../third-party/node-tree-sitter/tree-sitter.d.ts:208
```